gcore/gdal_priv.h

```diff
@@ -60,6 +60,7 @@ class GDALRelationship;
 
 #include "gdal.h"
 #include "gdal_frmts.h"
+#include "gdalsubdatasetinfo.h"
```

Comment:
this header file should be added in target_public_header() in gcore/CMakeLists.txt